pkg/server/server.go

```diff
@@ -393,7 +395,8 @@ func NewServerWithOpts(opts ...OpenFGAServiceV1Option) (*Server, error) {
 			zap.Uint32("Threshold", s.dispatchThrottlingThreshold),
 		)
 
-		dispatchThrottlingCheckResolver := graph.NewDispatchThrottlingCheckResolver(dispatchThrottlingConfig)
+		s.dispatchThrottler = throttler.NewDispatchThrottler(dispatchThrottlingConfig)
+		dispatchThrottlingCheckResolver := graph.NewDispatchThrottlingCheckResolver(s.dispatchThrottler)
```

Comment:
Won't block the PR for this, but i'd like to point out because it concerns me. The number of fields that we've introduced in the server is getting quite large. So is the server constructor, which for me is very hard to read.

```go
dispatchThrottler               *throttler.DispatchThrottler
dispatchThrottlingCheckResolver *graph.DispatchThrottlingCheckResolver
cachedCheckResolver    *graph.CachedCheckResolver
checkResolver graph.CheckResolver
typesystemResolver     typesystem.TypesystemResolverFunc
````

In reality, the concrete objects that need to be accessible by the APIs are:
- the model cache (almost every API needs this)
- the check cache (Check and ListObjects)
- with this PR, the throttling queue(s) (Check, ListObjects, and maybe more APIs in the future)

If y'all agree that this is something that we can improve, can you please open a ticket? Then we can talk about how to make it better.